kernel/runtime/cxx/runtime/futex.cc

```diff
@@ -167,6 +169,85 @@ static int futex_wake(
+                        DVAR(curr->load(std::memory_order_relaxed)->ec), DVAR(curr->load(std::memory_order_relaxed)->queueNext), DVARhex(uaddr) );
+            curr = &curr->load(std::memory_order_relaxed)->queueNext;
+        }
+
+        if (curr->load(std::memory_order_relaxed) == nullptr){
+          MLOG_DETAIL(mlog::app, "requeue end (end of queue reached)" );
+          return 0;
+        }
+
+        if (queue[hashFrom].queueTail == &curr->load(std::memory_order_relaxed)->queueNext) {
+            // own element is the tail of list:
+            // move tail to the element that points to our element
+            MLOG_DETAIL(mlog::app, "last elem in queue");
+            queue[hashFrom].queueTail = curr;
+        }
+        MLOG_DETAIL(mlog::app, uaddr, "Found entry" );
+        auto entry = curr->load(std::memory_order_relaxed);
+        curr->store(entry->queueNext, std::memory_order_relaxed);
+        auto ec = entry->ec;
+        entry->queueNext.store(reinterpret_cast<FutexQueueElem*>(1ul)); // mark as removed
+
+        MLOG_DETAIL(mlog::app, "Wake EC", DVAR(ec), DVAR(uaddr) );
+        mythos::syscall_signal(ec);
+    }
+
+    // If there are more than val waiters, then the remaining
+    // waiters are removed from the wait queue of the source futex at uaddr and added to the wait queue  of
+    // the  target  futex  at  uaddr2.  The val2 argument specifies an upper limit on the number of waiters
+    // that are requeued to the futex at uaddr2.
+    {
+      uint32_t hashTo = hash_futex(uaddr2);
+      mythos::Mutex::Lock guard2(queue[hashTo].readLock);
+      MLOG_DETAIL(mlog::app, "requeue waiters", DVAR(hashTo) );
+      for (int i = 0; i < val2; i++) {
+          while (curr->load(std::memory_order_relaxed) && curr->load(std::memory_order_relaxed)->uaddr != uaddr) {
+              MLOG_DETAIL(mlog::app, "Skip entry", DVARhex(curr->load(std::memory_order_relaxed)->uaddr),
+                          DVAR(curr->load(std::memory_order_relaxed)->ec), DVAR(curr->load(std::memory_order_relaxed)->queueNext), DVARhex(uaddr) );
+              curr = &curr->load(std::memory_order_relaxed)->queueNext;
+          }
+
+          if (curr->load(std::memory_order_relaxed) == nullptr){
+            MLOG_DETAIL(mlog::app, "requeue end (end of queue reached)" );
+            return 0;
+          }
+
+          if (queue[hashFrom].queueTail == &curr->load(std::memory_order_relaxed)->queueNext) {
+              // own element is the tail of list:
+              // move tail to the element that points to our element
+              MLOG_DETAIL(mlog::app, "last elem in queue");
+              queue[hashFrom].queueTail = curr;
+          }
+          MLOG_DETAIL(mlog::app, uaddr, "Found entry", DVARhex(curr) );
+          auto entry = curr->load(std::memory_order_relaxed);
+          curr->store(entry->queueNext, std::memory_order_relaxed);
+
+          // adapt entry
+          entry->queueNext.store(nullptr);
+          entry->uaddr = uaddr2;
```

Comment:
Same with removing the entry ...

Reply:
those are only 4 lines of code... I swear, if I need this code part another time, I will move it to an extra function :D